builtins: implement ST_GeomFromGeoHash/ST_Box2DFromGeoHash

otan opened the pull request:
Resolves #48806

Release note (sql change): Implemented the ST_GeomFromGeoHash and
ST_Box2DFromGeoHash builtins.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53162)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 6 of 6 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53162#-:-MFFjwnHEJrnf70rmZkQ:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/parse.go, line 253 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53162#-MFFj8oK7-e7qGd-aipi:-MFFj8oK7-e7qGd-aipj:bocklhj) ([raw file](https://github.com/cockroachdb/cockroach/blob/95b9585700e9b62e80d9ba0a54f79595688307fa/pkg/geo/parse.go#L253)):*
> ```Go
> 	// If precision is more than the length of the geohash
> 	// or if precision is less than 0 then set
> 	// precision equal to length of geohash.
> ```

What happens if we don't do this? is there a test that covers this?
Should we have a `parse_test.go` so that we're not relying on logic_tests for test coverage of such geo functions?


<!-- Sent from Reviewable.io -->

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53162)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)

___
*[pkg/geo/parse.go, line 253 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53162#-MFFj8oK7-e7qGd-aipi:-MFGQyfm7_LZa3C-7csW:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/95b9585700e9b62e80d9ba0a54f79595688307fa/pkg/geo/parse.go#L253)):*
<details><summary><i>Previously, sumeerbhola wrote…</i></summary><blockquote>

What happens if we don't do this? is there a test that covers this?
Should we have a `parse_test.go` so that we're not relying on logic_tests for test coverage of such geo functions?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

otan committed:
builtins: implement ST_GeomFromGeoHash/ST_Box2DFromGeoHash

Release note (sql change): Implemented the ST_GeomFromGeoHash and
ST_Box2DFromGeoHash builtins.

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53162#-:-MFGahON7bAYfQju-3Ef:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @sumeerbhola)



<!-- Sent from Reviewable.io -->
